lsp/nls/src/files.rs

```diff
@@ -31,10 +45,13 @@ pub fn handle_open(server: &mut Server, params: DidOpenTextDocumentParams) -> Re
             content: &params.text_document.text,
         },
     );
-    let file_id = server.cache.add_string(
-        params.text_document.uri.to_file_path().unwrap(),
-        params.text_document.text,
-    );
+    if params.text_document.uri.scheme() != "file" {
+        bail!("nls only supports file:// schemes");
```

Comment:
If I understand right, this only produces the correct error variant `SchemeNotSupported` because that currently is the only variant with a `From` conversion from a single string. That seems rather fragile.

Reply:
Oops, that was left over from an earlier change. The scheme checking is now only in the utility function.